### PR TITLE
Check if ament index resource exists before accessing

### PR DESCRIPTION
**Pull request by jacquelinekay**
This changes the behavior of `get_rmw_typesupport` slightly so that it returns an empty list rather than erroring if no typesupport is found for an rmw_implementation/language label.

Connects to ros2/rclpy#21

### CODE REVIEW

File: rmw/cmake/get_rmw_typesupport.cmake [hunk 1]
```diff
@@ -36,6 +36,11 @@ function(get_rmw_typesupport var rmw_implementation)
...
     set(resource_type "${resource_type}_${ARG_LANGUAGE}")
   endif()
-  ament_index_get_resource(resource "${resource_type}" "${rmw_implementation}")
-  set(${var} "${resource}" PARENT_SCOPE)
+  ament_index_has_resource(has_resource "${resource_type}" "${rmw_implementation}")
+  if(${has_resource})
+    ament_index_get_resource(resource "${resource_type}" "${rmw_implementation}")
+    set(${var} "${resource}" PARENT_SCOPE)
+  else()
+    set(${var} " " PARENT_SCOPE)
```
Comment (dirk-thomas) [inline rmw/cmake/get_rmw_typesupport.cmake]: Why a space?

Comment (jacquelinekay) [inline rmw/cmake/get_rmw_typesupport.cmake]: because I was thinking of how we do empty string compare in CMake (with a trailing space) and got mixed up

Comment (dirk-thomas) [inline rmw/cmake/get_rmw_typesupport.cmake]: Also only set the `resource` variable in the else case and then do the set in parent scope only once outside of the conditional block.

Comment (jacquelinekay) [inline rmw/cmake/get_rmw_typesupport.cmake]: Like in 65f7c70?

**Comment by dirk-thomas** [comment]
LGTM.

**Pull request merged**